File: Chapter 04/Python/shared/log_manager.py
from datetime import datetime

class LogManager:
    """
    SHARED UTILITY.
    Centralizes logging to ensure consistent formatting across the chapter.
    """
    
    @staticmethod
    def info(context: str, message: str) -> None:
        """
        Logs a formatted message with a timestamp and context.
        
        Args:
            context: The class or module originating the log.
            message: The message string using {0}, {1} style placeholders.
            *args: Variable arguments to fill the placeholders.
        """
        timestamp = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
        print(f"[{timestamp}] [INFO] [{context}] {message}")

File: Chapter 04/Java/src/main/java/com/grokkingsoftwarearchitecture/chapter04/shared/LogManager.java
package com.grokkingsoftwarearchitecture.chapter04.shared;

import java.util.logging.Logger;
import java.util.logging.Level;

/**
 * SHARED UTILITY.
 * Centralizes the creation of loggers to ensure consistent 
 * formatting and level control across all architectural layers.
 */
public final class LogManager {
    private LogManager() {} // Prevent instantiation

    /**
     * Standardized way to retrieve a logger for any class.
     */
    public static Logger getLogger(Class<?> clazz) {
        return Logger.getLogger(clazz.getName());
    }

    /**
     * Utility for a standardized info log to keep the Core clean.
     */
    public static void info(Class<?> clazz, String message, Object... params) {
        getLogger(clazz).log(Level.INFO, message, params);
    }
}